### PR TITLE
fix: add coffeescript support to migration

### PR DESCRIPTION
**Pull request by ImCesar**
We have not dropped support for coffeescript so we still need to account for it in the migration rename step

- Closes [UNIFY-1139](https://cypress-io.atlassian.net/browse/UNIFY-1139)

### PR Tasks

- [x] Have tests been added/updated?
- [ ] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
- [ ] Have new configuration options been added to the [`cypress.schema.json`](https://github.com/cypress-io/cypress/blob/develop/cli/schema/cypress.schema.json)?

### CODE REVIEW

File: system-tests/projects/migration-e2e-coffeescript/expected-cypress.config.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+const { defineConfig } = require('cypress')
```
Comment (lmiller1990) [inline system-tests/projects/migration-e2e-coffeescript/expected-cypress.config.js:1]: Nit: Should the `support/index.ts` above be `support.coffee`? Same with plugins. This is a coffeescript project, after all.

File: system-tests/projects/migration-e2e-coffeescript/expected-cypress.config.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+module.exports = defineConfig({
+  e2e: {
+    // We've imported your old cypress plugins here.
+    // You may want to clean this up later by importing these.
+    setupNodeEvents (on, config) {
+      return require('./cypress/plugins/index.js')(on, config)
```
Comment (lmiller1990) [inline system-tests/projects/migration-e2e-coffeescript/expected-cypress.config.js:8]: I wonder if this should be `plugins/index.coffee`?

**Review by lmiller1990 — COMMENTED** [review]
Minor comments

File: system-tests/projects/migration-e2e-coffeescript/expected-cypress.config.js (hunk 1 above)
Comment (ImCesar) [reply system-tests/projects/migration-e2e-coffeescript/expected-cypress.config.js:1]: Good catch, testing for this case showed that we needed to update the regex there as well

**Review by flotwig — APPROVED** [review]
This fixes the migration. There seems to be another bug, `.coffee` specs are not shown on the specs list: 
image

@ImCesar I can open a new bug for this if you don't wanna fix it in this PR.

**Comment by ImCesar** [comment]
> This fixes the migration. There seems to be another bug, `.coffee` specs are not shown on the specs list: image
> 
> @ImCesar I can open a new bug for this if you don't wanna fix it in this PR.

Yah lets make another ticket for that because that's in a completely different part of the code

**Comment by lmiller1990** [comment]
@flotwig what is your `specPattern`? The default is `cypress/e2e/**/*.cy.{js,jsx,ts,tsx}`. Could that be why your coffee spec isn't showing?

Docs: https://www.notion.so/cypressdx/testFiles-specPattern-glob-patterns-295d545a0af942ebb5b71eda2a6d422b#0a94379943b248b6a2fac39f5e541410

**Review by lmiller1990 — APPROVED** [review]

**Comment by flotwig** [comment]
ah yeah, we probably need to expand that `specPattern` to include `.coffee` files if they are present in the project.

**Comment by lmiller1990** [comment]
I will follow up internally!

File: packages/launchpad/cypress/e2e/migration.cy.ts [hunk 3]
```diff
@@ -639,7 +677,7 @@ describe.skip('component testing migration - defaults', () => {
   })
 })
 
-describe('Migration', { viewportWidth: 1200 }, () => {
+describe('Migration', { viewportWidth: 1200, retries: { openMode: 2, runMode: 2 } }, () => {
```
Comment (brian-mann) [inline packages/launchpad/cypress/e2e/migration.cy.ts:680]: You could just do `retries: 2` here, and not need to specify the values for each mode

Comment (lmiller1990) [reply packages/launchpad/cypress/e2e/migration.cy.ts:680]: @ImCesar can you make this change? Thanks!

**Pull request merged**